lib/ruby_lsp/requests/document_highlight.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # The [document highlight](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentHighlight)
+    # informs the editor all relevant elements of the currently pointed item for highlighting. For example, when
+    # the cursor is on the `F` of the constant `FOO`, the editor should identify other occurences of `FOO`
+    # and highlight them.
+    #
+    # For writable elements like constants or variables, their read/write occurrences should be highlighted differently.
+    # This is achieved by sending different "kind" attributes to the editor (2 for read and 3 for write).
+    #
+    # # Example
+    #
+    # ```ruby
+    # FOO = 1 # should be highlighted as "write"
+    #
+    # def foo
+    #   FOO # should be highlighted as "read"
+    # end
+    # ````
+    class DocumentHighlight < BaseRequest
+      WRITE_KIND = 3
```

Comment:
Let's use the constants already exposed by the [language server gem](https://github.com/mtsmfm/language_server-protocol-ruby/blob/main/lib/language_server/protocol/constant/document_highlight_kind.rb).